src/main/scala/com/cognite/spark/datasource/CdpRelation.scala

```diff
@@ -124,5 +125,20 @@ abstract class CdpRelation[T <: Product: Decoder](config: RelationConfig, shortN
   def delete(rows: Seq[Row]): IO[Unit] =
     throw new IllegalArgumentException(
       s"""$shortName does not support the "onconflict" option "delete".""")
+}
+
+trait InsertSchema {
+  val insertSchema: StructType
+}
+
+trait UpsertSchema {
+  val upsertSchema: StructType
+}
+
+trait UpdateSchema {
+  val updateSchema: StructType
+}
 
+abstract class DeleteSchema {
```

Comment:
While tempting to use an abstract class to get a default value it's a bit annoying with the single inheritance.
I'd perhaps rather just define the value once in an object or something and do `val deleteSchema = DeleteSchema.withOnlyId` in the classes implementing it.

This is also ok though, we can always change it later if the inheritance thing becomes an issue.

Reply:
Good point, I think I'll keep it as this for now, since it can as you said always be changed later.